docs/content/topics/project-developers/events.md

```diff
@@ -9,4 +9,236 @@ aliases:
+
+In this document, we'll look at:
+
+  * The [structure of an event]
+  * How to [configure event subscriptions] on a project
+  * How to [handle events] in a Brigade script.
+
+[Gateways]: /topics/operators/gateways
+[Project]: /topics/project-developers/projects
+[structure of an event]: #event-structure
+[configure event subscriptions]: #event-subscriptions
+[handle events]: #handling-events
+
+## Event Structure
+
+A Brigade event is defined primarily by its source and type values. These two
+must be provided every time an event is emitted into Brigade. However, there
+are plenty of other fields that event creators can harness to add additional
+context and information for utilization in handlers.
+
+The full list of fields on an Event is:
+
+  * [Source](#source)
+  * [Type](#type)
+  * [Payload](#payload)
+  * [ProjectID](#project-id)
+  * [Labels](#labels)
+  * [Qualifiers](#qualifiers)
+  * [Short Title](#short-title)
+  * [Long Title](#long-title)
+  * [Source State](#source-state)
+  * [Summary](#summary)
+
+Each field is reviewed in depth below.
+
+### Source
+
+An event's Source can be thought of as its domain. For developers writing their
+own gateway, the rule of thumb to avoid clashes is to use a URI one controls.
+This means leading with one's own domain or the URL for something else one
+owns, like the URL for a GitHub repo. As an example, events emitted from
+Brigade's [GitHub Gateway] use a source value of `brigade.sh/github`.
+
+### Type
+
+The Type of an event is used to specify increased granularity of the event's
+role/purpose as it relates to the source. For example, an event emitted by
+the GitHub Gateway with Type `push` signifies that a push event has occurred
+on a particular GitHub repo.
+
+### Payload
+
+An event Payload can be used to send input that may be utilized when handling
+the event, but otherwise opaque to Brigade itself. One example would be sending
+the original GitHub push event payload on the corresponding GitHub gateway
+event, so that these additional details may be parsed in a project's Brigade
+script.
+
+### Project ID
+
+Although not normally used by general-purpose gateways, a Project ID value may
+be set on an event. In such cases, the event will _only_ be eligible for
+receipt by the project indicated by this fields' value. As an example, the
+`brig project create --project <id>` command emits an event with this field
+set to the supplied project ID value.
+
+### Labels
+
+Projects can choose to utilize Labels for filtering purposes. The labels on a
+project's event subscription do not need to exactly match the labels on an
+Event in order for the project to receive it. Labels can optionally be used to
+narrow an event subscription's scope by selecting only events that are labeled
+in a particular way. For example, a label of `branch: main` might be used on an
+event subscription to limit inbound GitHub events to only those that occur on
+the `main` branch of a given repo.
```

Comment:
This particular example might be dicey because the github gateway doesn't actually use such a label and that could confuse people.

I have looked into adding that label, but there are a lot of events the gateway deals with that are only specific to the repo and _not_ specific to any particular commit, branch, tag, etc...

I haven't ruled out adding this tag in cases where it is applicable, but I worry that if some events have this tag and others don't, it may really complicate things on the subscription end.

Reply:
So an example that's more hypothetical might cause less confusion.

Or... a real example that isn't "public" yet would involve my prototype Slack gateway which labels all events with channel of origin. So you can optionally ignore the label and get all events for the workspace or specify the label and get events only from a specific channel.

Reply:
Ok, usin' the Slack gateway example -- best wrap that gateway up! 😜

Reply:
I'm on it!